src/main/java/ai/elimu/web/SignOnControllerDiscord.java

```diff
@@ -0,0 +1,179 @@
+            clientSecret = ConfigHelper.getProperty("discord.api.secret");
+            baseUrl = "https://" + request.getServerName();
+        }
+
+
+        oAuth20Service = new ServiceBuilder(clientId)
+                .apiSecret(clientSecret)
+                .defaultScope("identify email") 
+                .callback(baseUrl + "/sign-on/discord/callback")
+                .build(DiscordApi.instance());
+
+        logger.info("Fetching the Authorization URL...");
+        String authorizationUrl = oAuth20Service.createAuthorizationUrlBuilder()
+                .state(secretState)
+                .build();
+        logger.info("Redirecting to the Authorization URL: " + authorizationUrl);
+        logger.info("Got the Authorization URL!");
+		
+        return "redirect:" + authorizationUrl;
+    }
+
+    @RequestMapping(value="/sign-on/discord/callback", method=RequestMethod.GET)
+    public String handleCallback(HttpServletRequest request) throws IOException, InterruptedException, ExecutionException {
+        logger.info("handleCallback");
+
+        String state = request.getParameter("state");
+        logger.debug("state: " + state);
+        if (!secretState.equals(state)) {
+            return "redirect:/sign-on?error=state_mismatch";
+        } else {
+            String code = request.getParameter("code");
+            logger.debug("verifierParam: " + code);
+            logger.info(code);
+            String responseBody = null;
+            
+            logger.info("Trading the Authorization Code for an Access Token...");
+            try {
+                OAuth2AccessToken accessToken = oAuth20Service.getAccessToken(code);
+                logger.debug("accessToken: " + accessToken);
+                logger.info("Got the Access Token!");
+                logger.info("Now let's go and ask for the protected resource...");
+                // Access the protected resource
+                OAuthRequest oAuthRequest = new OAuthRequest(Verb.GET, PROTECTED_RESOURCE_URL);
+                oAuth20Service.signRequest(accessToken, oAuthRequest);
+                Response response = oAuth20Service.execute(oAuthRequest);
+                responseBody = response.getBody();
+                logger.info("response.getCode(): " + response.getCode());
+                logger.info("response.getBody(): " + responseBody);
+            } catch (IOException | InterruptedException | ExecutionException ex) {
+                logger.error(ex);
+                return "redirect:/sign-on?error=" + ex.getMessage();
+            }
+
+            JSONObject jsonObject = new JSONObject(responseBody);
+            logger.info("jsonObject: " + jsonObject);
+
+            Contributor contributor = new Contributor();
+            if (jsonObject.has("email") && !jsonObject.isNull("email")) {
+                contributor.setEmail(jsonObject.getString("email"));
+            }
+            if (jsonObject.has("username")) {
```

Comment:
Can we also put "`<username>`" + "`#<discriminator>`" as a Contributor's firstName + lastName?